src/Voron/Exceptions/VoronUnrecoverableErrorException.cs

```diff
@@ -6,11 +6,26 @@
 
 using System;
 using System.Runtime.ExceptionServices;
+using Voron.Impl;
 
 namespace Voron.Exceptions
 {
     public class VoronUnrecoverableErrorException : Exception
     {
+        public static void Raise(LowLevelTransaction tx, string message)
+        {
+            try
+            {
+                var lastTxState = tx.GetTxState();
+                tx.MarkTransactionAsFailed();
+                throw new VoronUnrecoverableErrorException($"txState: {lastTxState}, message: {message}");
```

Comment:
The tx state will be mostly for debugging / troubleshooting so let's put it _after_ the actual message (`$"{message}. Current tx state: {lastTxState}"`

Reply:
done